server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -2336,7 +2338,9 @@ protected void commitIndexWriter(final IndexWriter writer, final Translog transl
                 commitData.put(MAX_UNSAFE_AUTO_ID_TIMESTAMP_COMMIT_ID, Long.toString(maxUnsafeAutoIdTimestamp.get()));
                 commitData.put(HISTORY_UUID_KEY, historyUUID);
                 if (softDeleteEnabled) {
-                    commitData.put(Engine.MIN_RETAINED_SEQNO, Long.toString(softDeletesPolicy.getMinRetainedSeqNo()));
+                    final Tuple<Long, Collection<RetentionLease>> retentionPolicy = softDeletesPolicy.getRetentionPolicy();
+                    commitData.put(Engine.MIN_RETAINED_SEQNO, Long.toString(retentionPolicy.v1()));
+                    commitData.put(Engine.RETENTION_LEASES, RetentionLease.encodeRetentionLeases(retentionPolicy.v2()));
```

Comment:
I don't know if you considered using "leases" from EngineConfig#retentionLeasesSupplier instead. Or maybe you want to have a consistent view of min_retained_seq_no and leases.

Reply:
Yes, it's to ensure a consistent view.

Reply:
I pushed 5c226d9b9ae852da9e4724783c106afbaa7a68e2./